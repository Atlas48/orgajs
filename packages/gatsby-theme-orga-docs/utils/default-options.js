const path = require('path')

const defaultOptions = {
  location: ['docs'],
  imageMaxWidth: 1380,
  components: {},
}

module.exports = options => {
  return {
    ...defaultOptions,
    ...options,
  }
}
