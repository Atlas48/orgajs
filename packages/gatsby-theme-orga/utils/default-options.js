const path = require('path')

const defaultOptions = {
  contentPath: 'content',
  filter: () => true,
  pagination: 0,
  columns: 2,
  indexPath: '/',
  imageMaxWidth: 1380,
  categoryIndexPath: category => `/${category}`,
  tagIndexPath: tag => `/:${tag}:`,
  postPath: ({ category, export_file_name }) =>
    path.resolve('/', category || '', export_file_name || ''),
}

module.exports = options => {
  return {
    ...defaultOptions,
    ...options,
  }
}
