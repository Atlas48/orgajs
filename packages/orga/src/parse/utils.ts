import { Node } from 'unist'
import { push } from '../node'
import { Lexer } from '../tokenize'
import { Parent, Token } from '../types'

export default (lexer: Lexer) => {
  const { peek, eat, save, restore } = lexer
  const collect = (stop: (n: Token) => boolean) => (container: Parent): Parent => {
    const token = peek()
    if (!token || stop(token)) return container
    eat()
    push(container)(token)
    return collect(stop)(container)
  }

  const skip = (predicate: (token: Token) => boolean): void => {
    const token = peek()
    if (token && predicate(token)) {
      eat()
      skip(predicate)
      return
    }
    return
  }

  const tryTo = <T extends Node>(
    parse: (lexer: Lexer) => T | undefined,
    action: (node: T) => void,
  ): boolean => {
    const savePoint = save()
    const node = parse(lexer)
    if (!node) {
      restore(savePoint)
      return false
    }
    action(node)
    return true
  }

  return {
    collect,
    skip,
    tryTo,
  }
}
