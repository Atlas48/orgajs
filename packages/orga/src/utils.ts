import { Token } from './types'
import { PhrasingContent } from 'orga';

const matchOperatorsRe = /[|\\{}()[\]^$+*?.]/g

export const escape = (str: string): string => {
  return str.replace(matchOperatorsRe, '\\$&')
}

export const isPhrasingContent = (token: Token): token is PhrasingContent => {
  return token.type.startsWith('text.')
    || token.type === 'footnote.reference'
    || token.type === 'link'
}
