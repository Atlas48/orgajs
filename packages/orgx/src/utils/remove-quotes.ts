export const removeQuotes = (text: string) =>
  text.trim().replace(/^["'](.+(?=["']$))["']$/, '$1')

export default removeQuotes
