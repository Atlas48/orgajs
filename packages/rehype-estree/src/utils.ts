export const removeQuotes = (text: string) =>
  text.trim().replace(/^["'](.+(?=["']$))["']$/, '$1')
