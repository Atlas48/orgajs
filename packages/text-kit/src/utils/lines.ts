import { Point, Position } from 'unist'
import { CoreAPI } from '../core'

export default <T extends CoreAPI>(core: T) => {
  const linePosition = (
    start: number | Point,
    end?: number | Point
  ): Position | undefined => {
    let result: Position | undefined

    if (typeof start === 'number') {
      result = linePosition(core.toPoint(start))
    } else {
      const s = core.bol(start.line)
      const e = core.eol(start.line)
      if (!s || !e) return undefined
      result = {
        start: s,
        end: e,
      }
    }

    if (end && result) {
      const endLR = linePosition(end)
      if (!endLR) return undefined
      result.end = endLR.end
    }

    return result
  }
  return {
    ...core,
    linePosition,
  }
}
