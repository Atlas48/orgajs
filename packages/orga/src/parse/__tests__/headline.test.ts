import { parse } from './utils'

it('should handle timestamp after headline', () => {
  expect(
    parse(
      `
* headline
<2019-08-19 Mon>--<2019-08-20 Tue>

Paragraph
`.trim()
    )
  ).toMatchSnapshot()
})

it('should handle nested headlines', () => {
  expect(
    parse(
      `
* #HEADLINE# 1
Paragraph
** #HEADLINE# 1.1
*** #HEADLINE# 1.1.1
content

** #HEADLINE# 1.2
* #HEADLINE# 2
** #HEADLINE# 2.2
`.trim()
    )
  ).toMatchSnapshot()
})
