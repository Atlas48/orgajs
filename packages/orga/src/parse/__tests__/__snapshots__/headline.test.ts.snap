// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should handle nested headlines 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "actionable": false,
          "children": Array [
            Object {
              "level": 1,
              "parent": [Circular],
              "position": Object {
                "end": Object {
                  "column": 2,
                  "line": 1,
                  "offset": 1,
                },
                "start": Object {
                  "column": 1,
                  "line": 1,
                },
              },
              "type": "stars",
            },
            Object {
              "parent": [Circular],
              "position": Object {
                "end": Object {
                  "column": 15,
                  "line": 1,
                  "offset": 14,
                },
                "start": Object {
                  "column": 3,
                  "line": 1,
                  "offset": 2,
                },
              },
              "type": "text.plain",
              "value": "#HEADLINE# 1",
            },
            Object {
              "parent": [Circular],
              "position": Object {
                "end": Object {
                  "column": 1,
                  "line": 2,
                  "offset": 15,
                },
                "start": Object {
                  "column": 15,
                  "line": 1,
                  "offset": 14,
                },
              },
              "type": "newline",
            },
          ],
          "content": "#HEADLINE# 1",
          "level": 1,
          "parent": [Circular],
          "position": Object {
            "end": Object {
              "column": 1,
              "line": 2,
              "offset": 15,
            },
            "start": Object {
              "column": 1,
              "line": 1,
            },
          },
          "type": "headline",
        },
        Object {
          "attributes": Object {},
          "children": Array [
            Object {
              "parent": [Circular],
              "position": Object {
                "end": Object {
                  "column": 10,
                  "line": 2,
                  "offset": 24,
                },
                "start": Object {
                  "column": 1,
                  "line": 2,
                  "offset": 15,
                },
              },
              "type": "text.plain",
              "value": "Paragraph",
            },
          ],
          "parent": [Circular],
          "position": Object {
            "end": Object {
              "column": 1,
              "line": 3,
              "offset": 25,
            },
            "start": Object {
              "column": 1,
              "line": 2,
              "offset": 15,
            },
          },
          "type": "paragraph",
        },
        Object {
          "children": Array [
            Object {
              "actionable": false,
              "children": Array [
                Object {
                  "level": 2,
                  "parent": [Circular],
                  "position": Object {
                    "end": Object {
                      "column": 3,
                      "line": 3,
                      "offset": 27,
                    },
                    "start": Object {
                      "column": 1,
                      "line": 3,
                      "offset": 25,
                    },
                  },
                  "type": "stars",
                },
                Object {
                  "parent": [Circular],
                  "position": Object {
                    "end": Object {
                      "column": 18,
                      "line": 3,
                      "offset": 42,
                    },
                    "start": Object {
                      "column": 4,
                      "line": 3,
                      "offset": 28,
                    },
                  },
                  "type": "text.plain",
                  "value": "#HEADLINE# 1.1",
                },
                Object {
                  "parent": [Circular],
                  "position": Object {
                    "end": Object {
                      "column": 1,
                      "line": 4,
                      "offset": 43,
                    },
                    "start": Object {
                      "column": 18,
                      "line": 3,
                      "offset": 42,
                    },
                  },
                  "type": "newline",
                },
              ],
              "content": "#HEADLINE# 1.1",
              "level": 2,
              "parent": [Circular],
              "position": Object {
                "end": Object {
                  "column": 1,
                  "line": 4,
                  "offset": 43,
                },
                "start": Object {
                  "column": 1,
                  "line": 3,
                  "offset": 25,
                },
              },
              "type": "headline",
            },
            Object {
              "children": Array [
                Object {
                  "actionable": false,
                  "children": Array [
                    Object {
                      "level": 3,
                      "parent": [Circular],
                      "position": Object {
                        "end": Object {
                          "column": 4,
                          "line": 4,
                          "offset": 46,
                        },
                        "start": Object {
                          "column": 1,
                          "line": 4,
                          "offset": 43,
                        },
                      },
                      "type": "stars",
                    },
                    Object {
                      "parent": [Circular],
                      "position": Object {
                        "end": Object {
                          "column": 21,
                          "line": 4,
                          "offset": 63,
                        },
                        "start": Object {
                          "column": 5,
                          "line": 4,
                          "offset": 47,
                        },
                      },
                      "type": "text.plain",
                      "value": "#HEADLINE# 1.1.1",
                    },
                    Object {
                      "parent": [Circular],
                      "position": Object {
                        "end": Object {
                          "column": 1,
                          "line": 5,
                          "offset": 64,
                        },
                        "start": Object {
                          "column": 21,
                          "line": 4,
                          "offset": 63,
                        },
                      },
                      "type": "newline",
                    },
                  ],
                  "content": "#HEADLINE# 1.1.1",
                  "level": 3,
                  "parent": [Circular],
                  "position": Object {
                    "end": Object {
                      "column": 1,
                      "line": 5,
                      "offset": 64,
                    },
                    "start": Object {
                      "column": 1,
                      "line": 4,
                      "offset": 43,
                    },
                  },
                  "type": "headline",
                },
                Object {
                  "attributes": Object {},
                  "children": Array [
                    Object {
                      "parent": [Circular],
                      "position": Object {
                        "end": Object {
                          "column": 8,
                          "line": 5,
                          "offset": 71,
                        },
                        "start": Object {
                          "column": 1,
                          "line": 5,
                          "offset": 64,
                        },
                      },
                      "type": "text.plain",
                      "value": "content",
                    },
                  ],
                  "parent": [Circular],
                  "position": Object {
                    "end": Object {
                      "column": 1,
                      "line": 7,
                      "offset": 73,
                    },
                    "start": Object {
                      "column": 1,
                      "line": 5,
                      "offset": 64,
                    },
                  },
                  "type": "paragraph",
                },
              ],
              "level": 3,
              "parent": [Circular],
              "position": Object {
                "end": Object {
                  "column": 1,
                  "line": 7,
                  "offset": 73,
                },
                "start": Object {
                  "column": 1,
                  "line": 4,
                  "offset": 43,
                },
              },
              "properties": Object {},
              "type": "section",
            },
          ],
          "level": 2,
          "parent": [Circular],
          "position": Object {
            "end": Object {
              "column": 1,
              "line": 7,
              "offset": 73,
            },
            "start": Object {
              "column": 1,
              "line": 3,
              "offset": 25,
            },
          },
          "properties": Object {},
          "type": "section",
        },
        Object {
          "children": Array [
            Object {
              "actionable": false,
              "children": Array [
                Object {
                  "level": 2,
                  "parent": [Circular],
                  "position": Object {
                    "end": Object {
                      "column": 3,
                      "line": 7,
                      "offset": 75,
                    },
                    "start": Object {
                      "column": 1,
                      "line": 7,
                      "offset": 73,
                    },
                  },
                  "type": "stars",
                },
                Object {
                  "parent": [Circular],
                  "position": Object {
                    "end": Object {
                      "column": 18,
                      "line": 7,
                      "offset": 90,
                    },
                    "start": Object {
                      "column": 4,
                      "line": 7,
                      "offset": 76,
                    },
                  },
                  "type": "text.plain",
                  "value": "#HEADLINE# 1.2",
                },
                Object {
                  "parent": [Circular],
                  "position": Object {
                    "end": Object {
                      "column": 1,
                      "line": 8,
                      "offset": 91,
                    },
                    "start": Object {
                      "column": 18,
                      "line": 7,
                      "offset": 90,
                    },
                  },
                  "type": "newline",
                },
              ],
              "content": "#HEADLINE# 1.2",
              "level": 2,
              "parent": [Circular],
              "position": Object {
                "end": Object {
                  "column": 1,
                  "line": 8,
                  "offset": 91,
                },
                "start": Object {
                  "column": 1,
                  "line": 7,
                  "offset": 73,
                },
              },
              "type": "headline",
            },
          ],
          "level": 2,
          "parent": [Circular],
          "position": Object {
            "end": Object {
              "column": 1,
              "line": 8,
              "offset": 91,
            },
            "start": Object {
              "column": 1,
              "line": 7,
              "offset": 73,
            },
          },
          "properties": Object {},
          "type": "section",
        },
      ],
      "level": 1,
      "parent": [Circular],
      "position": Object {
        "end": Object {
          "column": 1,
          "line": 8,
          "offset": 91,
        },
        "start": Object {
          "column": 1,
          "line": 1,
        },
      },
      "properties": Object {},
      "type": "section",
    },
    Object {
      "children": Array [
        Object {
          "actionable": false,
          "children": Array [
            Object {
              "level": 1,
              "parent": [Circular],
              "position": Object {
                "end": Object {
                  "column": 2,
                  "line": 8,
                  "offset": 92,
                },
                "start": Object {
                  "column": 1,
                  "line": 8,
                  "offset": 91,
                },
              },
              "type": "stars",
            },
            Object {
              "parent": [Circular],
              "position": Object {
                "end": Object {
                  "column": 15,
                  "line": 8,
                  "offset": 105,
                },
                "start": Object {
                  "column": 3,
                  "line": 8,
                  "offset": 93,
                },
              },
              "type": "text.plain",
              "value": "#HEADLINE# 2",
            },
            Object {
              "parent": [Circular],
              "position": Object {
                "end": Object {
                  "column": 1,
                  "line": 9,
                  "offset": 106,
                },
                "start": Object {
                  "column": 15,
                  "line": 8,
                  "offset": 105,
                },
              },
              "type": "newline",
            },
          ],
          "content": "#HEADLINE# 2",
          "level": 1,
          "parent": [Circular],
          "position": Object {
            "end": Object {
              "column": 1,
              "line": 9,
              "offset": 106,
            },
            "start": Object {
              "column": 1,
              "line": 8,
              "offset": 91,
            },
          },
          "type": "headline",
        },
        Object {
          "children": Array [
            Object {
              "actionable": false,
              "children": Array [
                Object {
                  "level": 2,
                  "parent": [Circular],
                  "position": Object {
                    "end": Object {
                      "column": 3,
                      "line": 9,
                      "offset": 108,
                    },
                    "start": Object {
                      "column": 1,
                      "line": 9,
                      "offset": 106,
                    },
                  },
                  "type": "stars",
                },
                Object {
                  "parent": [Circular],
                  "position": Object {
                    "end": Object {
                      "column": 18,
                      "line": 9,
                      "offset": 123,
                    },
                    "start": Object {
                      "column": 4,
                      "line": 9,
                      "offset": 109,
                    },
                  },
                  "type": "text.plain",
                  "value": "#HEADLINE# 2.2",
                },
              ],
              "content": "#HEADLINE# 2.2",
              "level": 2,
              "parent": [Circular],
              "position": Object {
                "end": Object {
                  "column": 18,
                  "line": 9,
                  "offset": 123,
                },
                "start": Object {
                  "column": 1,
                  "line": 9,
                  "offset": 106,
                },
              },
              "type": "headline",
            },
          ],
          "level": 2,
          "parent": [Circular],
          "position": Object {
            "end": Object {
              "column": 18,
              "line": 9,
              "offset": 123,
            },
            "start": Object {
              "column": 1,
              "line": 9,
              "offset": 106,
            },
          },
          "properties": Object {},
          "type": "section",
        },
      ],
      "level": 1,
      "parent": [Circular],
      "position": Object {
        "end": Object {
          "column": 18,
          "line": 9,
          "offset": 123,
        },
        "start": Object {
          "column": 1,
          "line": 8,
          "offset": 91,
        },
      },
      "properties": Object {},
      "type": "section",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 18,
      "line": 9,
      "offset": 123,
    },
    "start": Object {
      "column": 1,
      "line": 1,
    },
  },
  "properties": Object {},
  "type": "document",
}
`;

exports[`should handle timestamp after headline 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "actionable": false,
          "children": Array [
            Object {
              "level": 1,
              "parent": [Circular],
              "position": Object {
                "end": Object {
                  "column": 2,
                  "line": 1,
                  "offset": 1,
                },
                "start": Object {
                  "column": 1,
                  "line": 1,
                },
              },
              "type": "stars",
            },
            Object {
              "parent": [Circular],
              "position": Object {
                "end": Object {
                  "column": 11,
                  "line": 1,
                  "offset": 10,
                },
                "start": Object {
                  "column": 3,
                  "line": 1,
                  "offset": 2,
                },
              },
              "type": "text.plain",
              "value": "headline",
            },
            Object {
              "parent": [Circular],
              "position": Object {
                "end": Object {
                  "column": 1,
                  "line": 2,
                  "offset": 11,
                },
                "start": Object {
                  "column": 11,
                  "line": 1,
                  "offset": 10,
                },
              },
              "type": "newline",
            },
          ],
          "content": "headline",
          "level": 1,
          "parent": [Circular],
          "position": Object {
            "end": Object {
              "column": 1,
              "line": 2,
              "offset": 11,
            },
            "start": Object {
              "column": 1,
              "line": 1,
            },
          },
          "type": "headline",
        },
        Object {
          "attributes": Object {},
          "children": Array [
            Object {
              "parent": [Circular],
              "position": Object {
                "end": Object {
                  "column": 35,
                  "line": 2,
                  "offset": 45,
                },
                "start": Object {
                  "column": 1,
                  "line": 2,
                  "offset": 11,
                },
              },
              "type": "text.plain",
              "value": "<2019-08-19 Mon>--<2019-08-20 Tue>",
            },
          ],
          "parent": [Circular],
          "position": Object {
            "end": Object {
              "column": 1,
              "line": 4,
              "offset": 47,
            },
            "start": Object {
              "column": 1,
              "line": 2,
              "offset": 11,
            },
          },
          "type": "paragraph",
        },
        Object {
          "attributes": Object {},
          "children": Array [
            Object {
              "parent": [Circular],
              "position": Object {
                "end": Object {
                  "column": 10,
                  "line": 4,
                  "offset": 56,
                },
                "start": Object {
                  "column": 1,
                  "line": 4,
                  "offset": 47,
                },
              },
              "type": "text.plain",
              "value": "Paragraph",
            },
          ],
          "parent": [Circular],
          "position": Object {
            "end": Object {
              "column": 10,
              "line": 4,
              "offset": 56,
            },
            "start": Object {
              "column": 1,
              "line": 4,
              "offset": 47,
            },
          },
          "type": "paragraph",
        },
      ],
      "level": 1,
      "parent": [Circular],
      "position": Object {
        "end": Object {
          "column": 10,
          "line": 4,
          "offset": 56,
        },
        "start": Object {
          "column": 1,
          "line": 1,
        },
      },
      "properties": Object {},
      "type": "section",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 10,
      "line": 4,
      "offset": 56,
    },
    "start": Object {
      "column": 1,
      "line": 1,
    },
  },
  "properties": Object {},
  "type": "document",
}
`;
